Lib/test/test_locale.py

```diff
@@ -499,27 +499,18 @@ def test_defaults_UTF8(self):
         else:
             orig_getlocale = None
 
-        orig_env = {}
-        try:
+        with os_helper.EnvironmentVarGuard() as env:
             for key in ('LC_ALL', 'LC_CTYPE', 'LANG', 'LANGUAGE'):
-                if key in os.environ:
-                    orig_env[key] = os.environ[key]
-                    del os.environ[key]
-
-            os.environ['LC_CTYPE'] = 'UTF-8'
-
-            with check_warnings(('', DeprecationWarning)):
-                self.assertEqual(locale.getdefaultlocale(), (None, 'UTF-8'))
-
-        finally:
-            for k in orig_env:
-                os.environ[k] = orig_env[k]
+                env.unset(key)
 
-            if 'LC_CTYPE' not in orig_env:
-                del os.environ['LC_CTYPE']
+            env['LC_CTYPE'] = 'UTF-8'
 
-            if orig_getlocale is not None:
-                _locale._getdefaultlocale = orig_getlocale
+            try:
+                with check_warnings(('', DeprecationWarning)):
+                    self.assertEqual(locale.getdefaultlocale(), (None, 'UTF-8'))
+            finally:
+                if orig_getlocale is not None:
+                    _locale._getdefaultlocale = orig_getlocale
```

Comment:
Maybe just in case, keep a huge `try-finally` and the environment guard inside that block:

```py
try:
	with os_helper.EnvironmentVarGuard() as env:
		...
finally:
	if orig_getlocale is not None:
		...
```

This would reduce the diff and ease additions in the future if we want to change the with-block.